src/Tickets/Commerce/Notice_Handler.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+namespace TEC\Tickets\Commerce;
+
+/**
+ * Notice Handler for managing Admin view notices.
+ *
+ * @since TBD
+ *
+ * @package TEC\Tickets\Commerce
+ */
+class Notice_Handler {
+	/**
+	 * Resets the cache with the key from `\Tribe_Admin_Notices->get_transients()` so that
+	 * we can display a given notice on the same page that we are triggering it.
+	 *
+	 * This is here because of a possible bug in common.
+	 *
+	 * @since TBD
+	 */
+	private function clear_request_cache() {
+		// Clear the existing notices cache.
+		$cache = tribe( 'cache' );
+		unset( $cache['transient_admin_notices'] );
+	}
+
+	/**
+	 * Fetches the expiration in seconds for the transient notice.
+	 *
+	 * @see   tribe_transient_notice()
+	 *
+	 * @since TBD
+	 *
+	 * @return int
+	 */
+	protected function get_expiration() {
+		return 10;
+	}
+
+	/**
+	 * Fetches the array of all messages available to display.
+	 *
+	 * @since TBD
+	 *
+	 * @return array[]
+	 */
+	public function get_messages() {
+		$messages = [
```

Comment:
This array is populated with PayPal messages, and it shouldn't be related to any gateway in particular. We should populate this array via filter (`tec_tickets_commerce_notice_messages `) on some file from the PayPal folder.